pkg/prm/prm.go

```diff
@@ -52,3 +75,153 @@ func (*Prm) cacheTool(tool *Tool) error {
 func (*Prm) getPuppetVersion() PuppetVersion {
 	return PuppetVersion{}
 }
+
+func (p *Prm) Get(toolDirPath string) (Tool, error) {
+	file := filepath.Join(toolDirPath, ToolConfigFileName)
+	_, err := p.AFS.Stat(file)
+	if os.IsNotExist(err) {
+		return Tool{}, fmt.Errorf("Couldn't find an installed tool at '%s'", toolDirPath)
+	}
+	i := p.readToolConfig(file)
+	if reflect.DeepEqual(i, Tool{}) {
+		return Tool{}, fmt.Errorf("Couldn't parse tool config at '%s'", file)
+	}
+	return i, nil
+}
+
+func (p *Prm) readToolConfig(configFile string) Tool {
+	file, err := p.AFS.ReadFile(configFile)
+	if err != nil {
+		log.Error().Msgf("unable to read tool config, %s", configFile)
+	}
+
+	var tool Tool
+
+	viper.SetConfigType("yaml")
+	err = viper.ReadConfig(bytes.NewBuffer(file))
+	if err != nil {
+		log.Error().Msgf("unable to read tool config, %s: %s", configFile, err.Error())
+		return Tool{}
+	}
+	err = viper.Unmarshal(&tool.Cfg)
+
+	if err != nil {
+		log.Error().Msgf("unable to parse tool config, %s", configFile)
+		return Tool{}
+	}
+
+	return tool
+}
+
+// List lists all templates in a given path and parses their configuration. Does
+// not return any errors from parsing invalid templates, but returns them as
+// debug log events
+func (p *Prm) List(toolPath string, toolName string) []ToolConfig {
+	log.Debug().Msgf("Searching %+v for tool configs", toolPath)
+	// Triple glob to match author/id/version/ToolConfigFileName
+	matches, _ := p.IOFS.Glob(toolPath + "/**/**/**/" + ToolConfigFileName)
+
+	var tmpls []ToolConfig
+	for _, file := range matches {
+		log.Debug().Msgf("Found: %+v", file)
+		i := p.readToolConfig(file)
```

Comment:
Should we do an empty check on this similar [to above](https://github.com/puppetlabs/prm/pull/33/files#diff-6273ef09fbedc97b5607eb2261f3272bcaa7b8c9c77ea601d27be24777bcdc09R86) in case there was an issue parsing the config and we get an NPE on the call to `i.Cfg`? Or is this another case where Go magically gives us `nils` back for uninitialised params? 😃

Reply:
We always get back an empty struct, regardless of error

Reply:
I was being silly, ignore me. Should've looked at the [Tool struct](https://github.com/puppetlabs/prm/pull/33/files#diff-076dd84dbf9c56f5f342e2239c10b9589bd37b32d927706b1c58475c780ed976R8-R13) and realised those params were there.